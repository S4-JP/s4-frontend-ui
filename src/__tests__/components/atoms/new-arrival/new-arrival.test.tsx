import React from 'react'
import { render } from '@testing-library/react'
import { NewArrival } from '../../../../index'

/**
 * type Props = {
 *   number?: number
 *   omit?: boolean
 *   small?: boolean
 * }
 */
describe('new-arrival のレンダリング確認', () => {
  test('オプションなしで設定した場合、初期値でレンダリングされる。', () => {
    const { getByTestId } = render(<NewArrival />)

    expect(getByTestId('atoms-na-wrap')).toMatchSnapshot()

    expect(getByTestId('atoms-na-wrap')).toHaveStyle(
      'width: 24px; height: 24px; font-size: 14px;'
    )
  })

  test('small=trueを設定した場合、設定値でレンダリングされる。', () => {
    const { getByTestId } = render(<NewArrival small={true} />)

    expect(getByTestId('atoms-na-wrap')).toHaveStyle(
      'width: 20px; height: 20px; font-size: 12px;'
    )
  })

  test('omit=trueを設定した場合、設定値でレンダリングされる。', () => {
    const { getByTestId } = render(<NewArrival small={true} omit={true} />)

    expect(getByTestId('atoms-na-wrap')).toHaveStyle(
      '  width: 16px; height: 16px; border-radius: 8px;'
    )
  })
})
