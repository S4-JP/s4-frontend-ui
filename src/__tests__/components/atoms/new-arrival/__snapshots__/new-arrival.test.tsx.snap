// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new-arrival のレンダリング確認 オプションなしで設定した場合、初期値でレンダリングされる。 1`] = `
<div
  class="sc-dwLEzm cNXLeJ"
  data-testid="atoms-na-wrap"
/>
`;
